controllers/datafoundation_client.go

```diff
@@ -0,0 +1,170 @@
+		predicate.NewPredicateFuncs(
+			func(res client.Object) bool {
+				return strings.HasPrefix(res.GetName(), ocsClientOperatorName)
+			},
+		),
+	)
+	controllerBuilder.
+		Owns(&corev1.Secret{}).
+		Owns(&opv1a1.ClusterServiceVersion{}, csvPredicates).
+		Owns(&ocsclient.StorageClient{})
+}
+
+func DFCAddToScheme(scheme *runtime.Scheme) {
+	utilruntime.Must(ocsclient.AddToScheme(scheme))
+}
+
+func dfcReconcile(offeringReconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering) (ctrl.Result, error) {
+	r := dataFoundationClientReconciler{}
+	r.initReconciler(offeringReconciler, offering)
+
+	if err := r.parseSpec(offering); err != nil {
+		return ctrl.Result{}, err
+	}
+	return r.reconcilePhases()
+}
+
+func (r *dataFoundationClientReconciler) initReconciler(reconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering) {
+	r.ManagedFusionOfferingReconciler = reconciler
+	r.offering = offering
+
+	r.storageclient.Name = "ocs-storageclient"
+	r.storageclient.Namespace = offering.Namespace
+}
+
+func (r *dataFoundationClientReconciler) parseSpec(offering *v1alpha1.ManagedFusionOffering) error {
+	r.Log.Info("Parsing ManagedFusionOffering Data Foundation Client spec")
+
+	isValid := true
+	onboardingTicket, found := offering.Spec.Config["onboarding_ticket"]
+	if !found {
+		r.Log.Error(
+			fmt.Errorf("missing field: onboarding_ticket"),
+			"an error occurred while parsing ManagedFusionOffering Data Foundation Client spec",
+		)
+		isValid = false
+	} else if onboardingTicket == "" {
+		r.Log.Error(
+			fmt.Errorf("onboarding_ticket is empty "),
+			"an error occurred while parsing ManagedFusionOffering Data Foundation Client spec",
+		)
+		isValid = false
+	}
+
+	dfstorageProviderEndpoint, found := offering.Spec.Config["data_foundation_provider_endpoint"]
+	if !found {
+		r.Log.Error(
+			fmt.Errorf("missing field: data_foundation_provider_endpoint"),
+			"an error occurred while parsing ManagedFusionOffering Data Foundation Client spec",
+		)
+		isValid = false
+	} else if dfstorageProviderEndpoint == "" {
+		r.Log.Error(
+			fmt.Errorf("data_foundation_provider_endpoint is empty "),
+			"an error occurred while parsing ManagedFusionOffering Data Foundation Client spec",
+		)
+		isValid = false
+	}
+
+	if !isValid {
+		r.Log.Info("parsing ManagedFusionOffering Data Foundation  Client spec failed")
+		return fmt.Errorf("invalid ManagedFusionOffering Data Foundation Client spec")
+	}
+	r.Log.Info("parsing ManagedFusionOffering Data Foundation Client spec completed successfuly")
+
+	r.dataFoundationClientSpec = dataFoundationClientSpec{
+		onboardingTicket:          onboardingTicket,
+		dfstorageProviderEndpoint: dfstorageProviderEndpoint,
+	}
+	return nil
+}
+
+func (r *dataFoundationClientReconciler) reconcilePhases() (ctrl.Result, error) {
+	if !r.offering.DeletionTimestamp.IsZero() {
+		return ctrl.Result{}, nil
+	} else {
+		if err := r.reconcileStorageClient(r.offering); err != nil {
```

Comment:
```suggestion
		if err := r.reconcileStorageClient(); err != nil {
```
Why are you passing offering to reconcileStorageClient? You can access it from `dataFoundationClientReconciler`

Reply:
Yes correct. Updated